cmd/lhsm-plugin-az-core/archive.go

```diff
@@ -17,30 +21,64 @@ type ArchiveOptions struct {
 	Credential    *azblob.SharedKeyCredential
 	Parallelism   uint16
 	BlockSize     int64
+	Pacer         util.Pacer
+	ExportPrefix  string
 }
 
 // persist a blob to the local filesystem
-func Archive(o ArchiveOptions) (int64, error){
-	ctx := context.TODO()
-	p := azblob.NewPipeline(o.Credential, azblob.PipelineOptions{})
+func Archive(o ArchiveOptions) (int64, error) {
+	archiveCtx := context.Background()
+	ctx, cancel := context.WithCancel(archiveCtx)
+	defer cancel()
+
+	p := util.NewPipeline(ctx, o.Credential, o.Pacer, azblob.PipelineOptions{})
+
+	//Get the blob URL
 	cURL, _ := url.Parse(fmt.Sprintf("https://%s.blob.core.windows.net/%s", o.AccountName, o.ContainerName))
 	containerURL := azblob.NewContainerURL(*cURL, p)
-	blobURL := containerURL.NewBlockBlobURL(o.BlobName)
+	dir, fileName := filepath.Split(o.BlobName)
+	blobURL := containerURL.NewBlockBlobURL(dir + o.ExportPrefix + fileName)
+
+	util.Log(pipeline.LogInfo, fmt.Sprintf("Archiving %s to %s.", o.BlobName, blobURL.String()))
 
 	// open the file to read from
-	file, _ := os.Open(o.SourcePath)
+	file, _ := os.Open(o.BlobName)
 	fileInfo, _ := file.Stat()
 	defer file.Close()
 
+	//Save owner, perm, group and acl info
 	total := fileInfo.Size()
 	meta := azblob.Metadata{}
+	owner := fmt.Sprintf("%d", fileInfo.Sys().(*syscall.Stat_t).Uid)
+	permissions := fmt.Sprintf("%o", fileInfo.Mode())
+	group := fmt.Sprintf("%d", fileInfo.Sys().(*syscall.Stat_t).Gid)
+	var acl string
+	modTime := fileInfo.ModTime().Format("2006-01-02 15:04:05 -0700")
 
-	meta["Permissions"] = fmt.Sprintf("%o", fileInfo.Mode())
-	meta["ModTime"] = fileInfo.ModTime().Format("2006-01-02 15:04:05 -0700")
-	meta["Owner"] = fmt.Sprintf("%d", fileInfo.Sys().(*syscall.Stat_t).Uid)
-	meta["Group"] = fmt.Sprintf("%d", fileInfo.Sys().(*syscall.Stat_t).Gid)
+	resp, err := blobURL.GetAccountInfo(context.Background())
+	if err != nil {
+		util.Log(pipeline.LogError, fmt.Sprintf("Archiving %s. Failed to account info: %s", o.BlobName, err.Error()))
+		return 0, err
+	}
```

Comment:
we can probably just check this once and not check it every time, future improvement